src/widget/wfindonmenu.cpp

```diff
@@ -0,0 +1,279 @@
+                    emit triggerBrowser(service, trackProperty, artist);
+                });
+        break;
+    }
+    default:
+        break;
+    }
+}
+
+void WFindOnMenu::addActionsAlbum(Service service,
+        const QString& albumName,
+        QMenu* pServiceMenu,
+        TrackProperty trackProperty) {
+    const QString prefixAction = composePrefixAction(trackProperty);
+    switch (service) {
+    case Service::SoundCloud: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, albumName),
+                this,
+                [this, service, trackProperty, albumName]() {
+                    emit triggerBrowser(service, trackProperty, albumName);
+                });
+        break;
+    }
+
+    case Service::LastFm: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, albumName),
+                this,
+                [this, service, trackProperty, albumName]() {
+                    emit triggerBrowser(service, trackProperty, albumName);
+                });
+        break;
+    }
+
+    case Service::Discogs: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, albumName),
+                this,
+                [this, service, trackProperty, albumName]() {
+                    emit triggerBrowser(service, trackProperty, albumName);
+                });
+        break;
+    }
+    default:
+        break;
+    }
+}
+
+void WFindOnMenu::addActionsTrackTitle(Service service,
+        const QString& trackTitle,
+        QMenu* pServiceMenu,
+        TrackProperty trackProperty) {
+    const QString prefixAction = composePrefixAction(trackProperty);
+    switch (service) {
+    case Service::SoundCloud: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, trackTitle),
+                this,
+                [this, service, trackProperty, trackTitle]() {
+                    emit triggerBrowser(service, trackProperty, trackTitle);
+                });
+        break;
+    }
+
+    case Service::LastFm: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, trackTitle),
+                this,
+                [this, service, trackProperty, trackTitle]() {
+                    emit triggerBrowser(service, trackProperty, trackTitle);
+                });
+        break;
+    }
+
+    case Service::Discogs: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, trackTitle),
+                this,
+                [this, service, trackProperty, trackTitle]() {
+                    emit triggerBrowser(service, trackProperty, trackTitle);
+                });
+        break;
+    }
+    default:
+        break;
+    }
+}
+
+void WFindOnMenu::openInBrowser(Service service,
+        TrackProperty trackProperty,
+        const QString& query) {
+    const QString serviceUrl = getServiceUrl(service, trackProperty);
+    QUrlQuery qurlquery;
```

Comment:
Repeating the type name in lowercase is not a good choice for naming a variable. Even the `Q` prefix is repeated.

Reply:
Oh, Okay. I will be more careful about not naming by repeating the type name. Thanks for this review. I will named it as "query".